src/System.Memory/src/System/MemoryExtensions.cs

```diff
@@ -17,6 +17,159 @@ namespace System
+        }
+
+        /// <summary>
+        /// Removes all trailing occurrences of a specified character.
+        /// </summary>
+        /// <param name="span">The source span from which the character is removed.</param>
+        /// <param name="trimChar">The specified character to look for and remove.</param>
+        public static ReadOnlySpan<char> TrimEnd(this ReadOnlySpan<char> span, char trimChar)
+        {
+            int end = span.Length - 1;
+            for (; end >= 0; end--)
+            {
+                if (span[end] != trimChar)
+                    break;
+            }
+            return span.Slice(0, end + 1);
+        }
+
+        /// <summary>
+        /// Removes all leading and trailing occurrences of a set of characters specified 
+        /// in a readonly span from the span.
+        /// </summary>
+        /// <param name="span">The source span from which the characters are removed.</param>
+        /// <param name="trimChars">The span which contains the set of characters to remove.</param>
+        public static ReadOnlySpan<char> Trim(this ReadOnlySpan<char> span, ReadOnlySpan<char> trimChars)
+        {
+            return span.TrimStart(trimChars).TrimEnd(trimChars);
+        }
+
+        /// <summary>
+        /// Removes all leading occurrences of a set of characters specified 
+        /// in a readonly span from the span.
+        /// </summary>
+        /// <param name="span">The source span from which the characters are removed.</param>
+        /// <param name="trimChars">The span which contains the set of characters to remove.</param>
+        public static ReadOnlySpan<char> TrimStart(this ReadOnlySpan<char> span, ReadOnlySpan<char> trimChars)
+        {
+            int start = 0;
+            for (; start < span.Length; start++)
+            {
+                int i = 0;
+                for (; i < trimChars.Length; i++)
+                {
+                    if (span[start] == trimChars[i])
+                        break;
+                }
+                if (i == trimChars.Length)
+                    break;
+            }
+            return span.Slice(start);
+        }
+
+        /// <summary>
+        /// Removes all trailing occurrences of a set of characters specified 
+        /// in a readonly span from the span.
+        /// </summary>
+        /// <param name="span">The source span from which the characters are removed.</param>
+        /// <param name="trimChars">The span which contains the set of characters to remove.</param>
+        public static ReadOnlySpan<char> TrimEnd(this ReadOnlySpan<char> span, ReadOnlySpan<char> trimChars)
+        {
+            int end = span.Length - 1;
+            for (; end >= 0; end--)
+            {
+                int i = 0;
+                for (; i < trimChars.Length; i++)
+                {
+                    if (span[end] == trimChars[i])
+                        break;
+                }
+                if (i == trimChars.Length)
```

Comment:
Nit: If you would like to save a few instructions here, I think `goto` would be perfectly acceptable.

Reply:
I do not think that the change you made is right. Do you see any failing tests? If not, it is a test hole you should fill as well.

Reply:
I think it needs to be something like:
```
            int start = 0;
            for (; start < span.Length; start++)
            {
                for (int i = 0; i < trimChars.Length; i++)
                {
                    if (span[start] == trimChars[i])
                        goto Next;
                }
                break;
        Next:
            }
            return span.Slice(start);
```

Reply:
> I do not think that the change you made is right. Do you see any failing tests?

Yep. The tests caught the bug. I hadn't updated the source dll before re-running the tests.